src/types/Union.ts

```diff
@@ -2,11 +2,16 @@ import * as t from 'io-ts'
 
 export interface Union<T extends string> extends t.UnionType<t.LiteralType<T>[], T, T, t.mixed> {}
 
-export const UnionOf: UnionOfType = <T extends string>(values: T[], name?: string): Union<T> => {
-    return t.union(
-        values.map(val => t.literal(val)),
-        name ? `Union:${name}` : 'Union'
-    )
+
+
+export const UnionOf: UnionOfType = <T extends string>(values: [T, T, ...T[]], name?: string): Union<T> => {
+    const [ value1, value2, ...valuesRest ] = values
```

Comment:
Can you please explain why this change `value1, value2, ...valuesRest`?

Reply:
UnionType in io-ts expects a tuple with at least 2 values, so here we're using same type signature and have to destructure it accordingly.